crates/optimism/evm/src/execute.rs

```diff
@@ -35,7 +37,10 @@ pub struct OpExecutorProvider<EvmConfig = OptimismEvmConfig> {
 impl OpExecutorProvider {
     /// Creates a new default optimism executor provider.
     pub fn optimism(chain_spec: Arc<ChainSpec>) -> Self {
-        Self::new(chain_spec, Default::default())
+        Self::new(
+            chain_spec.clone(),
+            OptimismEvmConfig::new(Arc::new(OpChainSpec { inner: (*chain_spec).clone() })),
+        )
     }
```

Comment:
Here maybe we could open a follow up issue to also use `OpChainSpec` directly in `OpExecutorProvider` instead of `ChainSpec`, wdyt?

Reply:
yeah that'd be ideal

Reply:
Opened an issue here https://github.com/paradigmxyz/reth/issues/10770